src/utils/application.ts

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Form, RetrievedApplication, State } from '@/types/application.types';
+
+function formatApplicationState(state: State) {
+  if (!state) return '';
+
+  const s = state.split('/')?.pop() || '';
+  return s.charAt(0).toUpperCase() + s.slice(1);
+}
+
+function isApplicationComplete(application: RetrievedApplication) {
+  if (!application.forms) return false;
+
+  return application.forms
+    .map((form) => form?.fields)
```

Comment:
you don't need form?.fields unless your form object itself if null within the array which I think is quite unlikely